spec/lib/id_token_spec.rb

```diff
@@ -45,9 +45,69 @@
   end
 
   describe '#as_jws_token' do
-    it 'returns claims encoded as JWT' do
-      jwt = JSON::JWT.decode_compact_serialized subject.as_jws_token, Doorkeeper::OpenidConnect.signing_key
-      expect(jwt.to_hash).to eq subject.as_json.stringify_keys
+    shared_examples 'a jws token' do
+      it 'returns claims encoded as JWT' do
+        jwt = JSON::JWT.decode_compact_serialized subject.as_jws_token, Doorkeeper::OpenidConnect.signing_key
+        expect(jwt.to_hash).to eq subject.as_json.stringify_keys
+      end
+    end
+
+    it_behaves_like 'a jws token'
+
+    context 'when encryption_method is EC' do
+      before do
+        Doorkeeper::OpenidConnect.configure do
+          jws_private_key <<-EOL
+-----BEGIN EC PRIVATE KEY-----
+MIHbAgEBBEF9VcxGjPKczrJlE1N3oEpZsauQfDXIjLeini7h4/3+DOKw2VWE4lCU
+rNJJL65EHT+2TriRg2xSb0l0rK/MAFAFraAHBgUrgQQAI6GBiQOBhgAEAeYVvbl3
+zZcFCdE+0msqOowYODjzeXAhjsZKhdNjGlDREvko3UFOw6S43g+s8bvVBmBz3fCo
+dEzFRYQqJVI4UFvFAYJ7GYeBm/Fb6liN53xGASdbRSzF34h4BDSVYzjtQc7I+1LK
+17fwwS3VfQCJwaT6zX33HTrhR4VoUEUJHKwR3dNs
+-----END EC PRIVATE KEY-----
+          EOL
+
+          encryption_method :ES512
+
+          resource_owner_from_access_token do |access_token|
+            User.find_by(id: access_token.resource_owner_id)
+          end
+
+          auth_time_from_resource_owner do |resource_owner|
+            resource_owner.current_sign_in_at
+          end
+
+          subject do |resource_owner|
+            resource_owner.id
+          end
+        end
+      end
+
+      it_behaves_like 'a jws token'
+    end
+
+    context 'when encryption_method is HMAC' do
+      before do
+        Doorkeeper::OpenidConnect.configure do
```

Comment:
Maybe add a helper method in the spec to avoid repeating the `subject` etc. settings? i.e. something like `setup_configuration(signing_key, signing_algorithm)`